metagpt/provider/openai_api.py

```diff
@@ -259,10 +263,14 @@ async def acompletion_batch_text(self, batch: list[list[dict]]) -> list[str]:
         return results
 
     def _update_costs(self, usage: dict):
-        if CONFIG.update_costs:
-            prompt_tokens = int(usage['prompt_tokens'])
-            completion_tokens = int(usage['completion_tokens'])
-            self._cost_manager.update_cost(prompt_tokens, completion_tokens, self.model)
+        if CONFIG.calc_usage:
+            try:
+                prompt_tokens = int(usage['prompt_tokens'])
+                completion_tokens = int(usage['completion_tokens'])
+                self._cost_manager.update_cost(prompt_tokens, completion_tokens, self.model)
+            except Exception as e:
+                logger.error("updating costs failed!", e)
+                CONFIG.calc_usage = False
```

Comment:
It is better to add comments for this setting change as there will be no log information for token and money cost in case of failure.